frontend/src/components/Map/index.tsx

```diff
@@ -1,9 +1,100 @@
-import React from 'react';
-import { MapContainer, TileLayer } from 'react-leaflet';
+import React, { useState, useEffect } from 'react';
+import {
+  MapContainer,
+  Marker,
+  Popup,
+  TileLayer,
+  useMapEvents
+} from 'react-leaflet';
+import { v4 } from 'uuid';
+import 'leaflet/dist/leaflet.css';
+import { lightLocationType } from '../../constants/develop';
+import { fetchData } from '../../fetch/requests';
+
+type latlngType = {
+  lat: number;
+  lng: number;
+};
+
+type boundsType = {
+  _northEast: {
+    lat: number;
+    lng: number;
+  };
+  _southWest: {
+    lat: number;
+    lng: number;
+  };
+};
 
 function Map() {
-  // TODO
-  // Function that defines coordinates on mouse click
+  const [bounds, setBounds] = useState<boundsType>({
+    _northEast: { lat: 54.82600799909498, lng: 38.64990234375001 },
+    _southWest: { lat: 45.62940492064501, lng: 22.456054687500004 }
+  });
+  const [center, setCenter] = useState<[number, number]>([
+    50.447731, 30.542721
+  ]);
+  const [zoom, setZoom] = useState<number>(6);
+  const [zoomPosition, setZoomPosition] = useState<latlngType>({
+    lat: 50.447731,
+    lng: 30.542721
+  });
+  const [locations, setLocations] = useState<lightLocationType[]>([]);
+
+  useEffect(() => {
+    async function onZoom() {
+      const url = `http://localhost:3001/api/locations/search/${zoom}/${JSON.stringify(
```

Comment:
put inside .env file (http://localhost:3001/api/) in frontend (variable name REACT_APP_API_URI)